### PR TITLE
Immunisation program demo

### PR DESCRIPTION
**Pull request by clemens-msupply**
This adds some demo for a potential immunisation program + encounters.

Since there are multiple encounters the UI needed an update (in order to create encounters of different types).

### CODE REVIEW

File: server/service/src/sync/program_schemas/routine_immunisation_6weeks_encounter_ui_schema.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            {
+              "type": "Control",
+              "scope": "#/properties/rotavirus/properties/vaccine"
+            }
+          ]
+        },
+        {
+          "type": "Category",
+          "label": "Diphtheria...",
+          "options": { "icon": "medicine" },
```
Comment (mark-prins) [inline server/service/src/sync/program_schemas/routine_immunisation_6weeks_encounter_ui_schema.json:42]: another suggestion.. 

Screenshot 2022-11-21 at 4 10 04 PM

added in a new svg and updated in this branch: https://github.com/openmsupply/open-msupply/compare/immunisation-program-demo...immunisation-program-demo-icon?expand=1

File: server/service/src/sync/program_schemas/routine_immunisation_6weeks_encounter.json [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+      "enum": ["Left Arm", "Right Arm", "Left Leg", "Right Leg"],
+      "type": "string"
+    },
+    "DiphtheriaTetanusPertussisPolioHepatitisBHaemophilus": {
+      "properties": {
+        "location": {
+          "$ref": "#/definitions/BodyLocation"
+        },
+        "vaccine": {
+          "$ref": "#/definitions/DiphtheriaTetanusPertussisPolioHepatitisBHaemophilusVaccine"
```
Comment (mark-prins) [inline server/service/src/sync/program_schemas/routine_immunisation_6weeks_encounter.json:14]: 😮

**Review by mark-prins — APPROVED** [review]
👍 very neat!

I have a concern about how this will integrate with dispensing later on - but let's not talk about that now!

A few components could be wider - the program name and end date are now not able to show all the data in them:

Screenshot 2022-11-21 at 2 17 23 PM

Screenshot 2022-11-21 at 2 17 31 PM
 
and the 'notes' element would be better with multiline enabled and also wider than this - if that's possible? maybe add some options to the ui_schema?

Screenshot 2022-11-21 at 2 16 40 PM


There's also a console warning `Warning: fragment with name EncounterRow already exists.` coming from `patient/api/operations.graphql` and `system/src/Encounter/api/operations.graphql` - both of them are defining an EncounterRow fragment

**Comment by clemens-msupply** [comment]
> +1 very neat!
> 
> I have a concern about how this will integrate with dispensing later on - but let's not talk about that now!

yes please let me know about it!

> A few components could be wider - the program name and end date are now not able to show all the data in them:

totally agree, this would need to be updated (another time). The whole thing was a bit quick and dirty and I actually made the note field small so that it look okish.

File: server/service/src/sync/program_schemas/routine_immunisation_6weeks_encounter_ui_schema.json (hunk 1 above)
Comment (clemens-msupply) [reply server/service/src/sync/program_schemas/routine_immunisation_6weeks_encounter_ui_schema.json:42]: Awesome!

File: client/packages/system/src/Patient/Components/EncounterSearchInput/EncounterSearchInput.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+export const EncounterSearchInput: FC<EncounterSearchInputProps> = ({
+  onChange,
+  width = 250,
+  value,
+  disabled = false,
+}) => {
+  const { data: programsEnrolments, isLoading: programsIsLoading } =
+    usePatient.document.programEnrolments();
+  const { data, isLoading } = usePatient.document.programEncounters(
```
[the remark this answers is not in the document]
Comment (clemens-msupply) [reply client/packages/system/src/Patient/Components/EncounterSearchInput/EncounterSearchInput.tsx]: Done [05397bc](https://github.com/openmsupply/open-msupply/pull/825/commits/05397bcb25c69469ee0eaecaf06f4176a7b51464)

File: client/packages/system/src/Patient/Components/EncounterSearchInput/EncounterSearchInput.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+interface EncounterSearchInputProps {
+  onChange: (type: EncounterRegistry) => void;
+  width?: number;
+  value: EncounterRegistry | null;
+  disabled?: boolean;
+}
+
+export const getEncounterOptionRenderer =
+  (): AutocompleteOptionRenderer<EncounterRegistry> => (props, node) => {
+    const name = node.encounter.name ?? '??';
```
[the remark this answers is not in the document]
Comment (clemens-msupply) [reply client/packages/system/src/Patient/Components/EncounterSearchInput/EncounterSearchInput.tsx]: Its actually the encounter registry entry and it technical doesn't need to have a name. I changed it to an empty string! [05397bc](https://github.com/openmsupply/open-msupply/pull/825/commits/05397bcb25c69469ee0eaecaf06f4176a7b51464)

File: client/packages/system/src/Patient/Encounter/DetailModal.tsx [hunk 5]
```diff
@@ -36,31 +34,24 @@ export const EncounterDetailModal: FC = () => {
...
-  const {
-    mutate: fetch,
-    isLoading,
-    data: registryData,
-  } = useEncounter.registry.byProgram();
-  const [documentRegistry, setDocumentRegistry] = useState<
-    EncounterDocumentRegistryFragment | undefined
-  >(undefined);
 
   const [data, setData] = useState<Encounter | undefined>(undefined);
```
[the remark this answers is not in the document]
Comment (clemens-msupply) [reply client/packages/system/src/Patient/Encounter/DetailModal.tsx]: changed it: [c9bd9a3](https://github.com/openmsupply/open-msupply/pull/825/commits/c9bd9a304bd24cccf2395a29c87f9afa712a5fad)

**Pull request merged**